airflow/operators/sensors.py

```diff
@@ -222,21 +222,98 @@ def poke(self, context):
         return count
 
 
+class NamedHivePartitionSensor(BaseSensorOperator):
+    """
+    Waits for a set of partitions to show up in Hive.
+
+    :param partition_names: List of fully qualified names of the
+        partitions to wait for. A fully qualified name is of the
+        form schema.table/pk1=pv1/pk2=pv2, for example,
+        default.users/ds=2016-01-01. This is passed as is to the metastore
+        Thrift client "get_partitions_by_name" method. Note that
+        you cannot use logical operators as in HivePartitionSensor.
+    :type partition_names: list<string>
+    :param metastore_conn_id: reference to the metastore thrift service
+        connection id
+    :type metastore_conn_id: str
+    """
+
+    template_fields = ('partition_names', )
+
+    @apply_defaults
+    def __init__(
+            self,
+            partition_names,
+            metastore_conn_id='metastore_default',
+            poke_interval=60*3,
+            *args,
+            **kwargs):
+        super(NamedHivePartitionSensor, self).__init__(
+            poke_interval=poke_interval, *args, **kwargs)
+
+        if isinstance(partition_names, basestring):
+            raise TypeError('partition_names must be an array of strings')
+
+        for partition_name in partition_names:
+            self.parse_partition_name(partition_name)
+
+        self.metastore_conn_id = metastore_conn_id
+        self.partition_names = partition_names
+        self.next_poke_idx = 0
+
+    def parse_partition_name(self, partition):
+        try:
+            schema, table_partition = partition.split('.')
+            table, partition = table_partition.split('/', 1)
+            return schema, table, partition
+        except ValueError as e:
+            raise ValueError('Could not parse ' + partition)
+
+    def poke(self, context):
```

Comment:
Seems like most of the logic here is shared with the existing sensor, does it make sense to pull a common superclass out?